openedx/core/djangoapps/password_policy/utils.py

```diff
@@ -0,0 +1,79 @@
+import pytz
+from datetime import datetime, timedelta
+from dateutil.parser import parse as parse_date
+
+from django.conf import settings
+from django.contrib import messages
+from django.contrib.auth.models import Group
+from django.utils.translation import ugettext as _
+
+from openedx.core.djangoapps.util.user_messages import PageLevelMessages, UserMessageType
+from openedx.core.djangoapps.password_policy.constants import PASSWORD_POLICY_COMPLIANT_USERS_GROUP_NAME
+from util.date_utils import strftime_localized, DEFAULT_SHORT_DATE_FORMAT
+from util.password_policy_validators import ValidationError, validate_password
+
+
+class NonCompliantPasswordException(Exception):
+    pass
+
+
+def check_password_policy_compliance(user, password, request=None):
+    config = settings.PASSWORD_POLICY_COMPLIANCE_ROLLOUT_CONFIG
+    if not (config and config.get('ENABLE_COMPLIANCE_CHECKING', False)):
+        return
+
+    # If the user is already known to be compliant, return.
+    if user.groups.filter(name=PASSWORD_POLICY_COMPLIANT_USERS_GROUP_NAME).exists():
+        return
+
+    is_compliant = is_password_compliant(user, password)
+    if is_compliant:
+        user.groups.add(Group.objects.get(name=PASSWORD_POLICY_COMPLIANT_USERS_GROUP_NAME))
+        return
+
+    deadline = get_enforcement_deadline_for_user(config, user)
+    if deadline is None:
+        return
+
+    now = datetime.now(pytz.UTC)
+    warning_days = config.get('DAYS_BEFORE_DEADLINE_TO_SHOW_WARNING', 0)
```

Comment:
I don't know what a good default is here, but disabling the warning seems like a dangerous one for a lazy or fat-fingered config-setter. Plus it's a different default than the default config has, which is odd.